jinahub/indexers/SimpleIndexer/simple_indexer.py

```diff
@@ -36,22 +35,21 @@ def __init__(
         self._docs = DocumentArrayMemmap(self.workspace + f'/{index_file_name}')
         self.default_traversal_paths = default_traversal_paths or ['r']
         self.default_top_k = default_top_k
-        if distance_metric == 'cosine':
-            self.distance = _cosine
-        elif distance_metric == 'euclidean':
-            self.distance = _euclidean
-        else:
-            raise ValueError('This distance metric is not available!')
+        self._distance = distance_metric
+        self._use_scipy = True
+        if distance_metric in ['cosine', 'euclidean', 'sqeuclidean']:
```

Comment:
Can u throw an error if distance is not supported?

Reply:
we do not have an exhaustive list. It depends on what `scipy` supports.

Reply:
These ones are the ones supported natively, there are others offered via `scipy`

Reply:
I changed the docstrings